src/mesh/unstructured/unstructuredMesh.cpp

```diff
@@ -105,4 +105,91 @@ UnstructuredMesh createSingleCellMesh(const Executor exec)
         boundaryMesh
     );
 }
+
+UnstructuredMesh create1DUniformMesh(
+    const Executor exec, const Vector leftBoundary, const Vector rightBoundary, const size_t nCells
+)
+{
+    // a 1D mesh in 3D space
+    // each cell has left and right faces
+    // the 1D mesh is aligned with the x coordinate of Cartesian coordinate sytstem
+
+    double meshSpacing = (rightBoundary[0] - leftBoundary[0]) / nCells;
```

Comment:
```suggestion
    scalar meshSpacing = 1.0 / nCells;
```
Even though `double` would work, please rather use `scalar` (which can be switched between `float` and `double` at compile time )